src/topology/instances/nnreal.lean

```diff
@@ -5,6 +5,8 @@ Authors: Johan Commelin
 -/
 import topology.algebra.infinite_sum
 import topology.algebra.group_with_zero
+import topology.order.lattice
+import algebra.order.lattice_group
```

Comment:
```suggestion
```

Reply:
In proving the `ℝ≥0∞` instance, I realized that there is a relatively straightforward proof and it can be generalized to this:
```lean
instance linear_order.topological_lattice {L : Type u} [topological_space L] [linear_order L]
  [order_closed_topology L] : topological_lattice L := sorry
```
so I have overhauled the PR to only provide this instance, which will give us the `topological_lattice` instances on all these sorts of things (`e.g., `ℝ`, `ℝ≥0`, `ℝ≥0∞`, `set.Icc a b`, `ereal`, `ℚ`, etc.).